src/cli.ts

```diff
@@ -1,13 +1,286 @@
+
+  // after
+  // $rootdir/Albums/My Album 1
+  // $rootdir/Albums/My Album 2
+  // $rootdir/Photos/Photos from 2008
+
+  const photosDir: string = `${rootDir}/Photos`;
+
+  // move the "Photos from $YEAR" directories to Photos/
+  _restructureIfNeeded(await glob(`${rootDir}/Photos from */`), photosDir);
+  
+  // move everythingg else to Albums/, so we end up with two top level folders
+  const fullSet: Set<string> = new Set(await glob(`${rootDir}/*/`));
+  const photoSet: Set<string> = new Set(await glob(`${rootDir}/Photos from */`));
+  photoSet.add(`${rootDir}/Photos`);
+  const everythingExceptPhotosDir: string[] = Array.from(
+    new Set([...fullSet].filter((x) => !photoSet.has(x)))
+  );
+  _restructureIfNeeded(everythingExceptPhotosDir, `${rootDir}/Albums`);
+}
+
+async function runFullMigration(
+  rootDir: string,
+  timeout: number,
+) {
+  // at least in my takeout, the Takeout folder contains a subfolder
+  // Takeout/Google Foto
+  // rootdir refers to that subfolder
+
+  rootDir = (await glob(`${rootDir}/Google*`))[0].replace(/\/+$/, '');
+  await restructureIfNeeded(rootDir);
+  await processPhotos(rootDir, timeout);
+  await processAlbums(rootDir, timeout);
+}
+
+const rewriteAllTags = command({
+  name: 'rewrite all tags for single file',
+  args: {
+    inFile: positional({
+      type: string,
+      displayName: 'in_file',
+      description: 'The path to your input file.',
+    }),
+    outFile: positional({
+      type: string,
+      displayName: 'out_file',
+      description: 'The path to your output location for the file.',
+    }),
+    timeout: option({
+      type: number,
+      defaultValue: () => 30000,
+      short: 't',
+      long: 'timeout',
+      description:
+        'Sets the task timeout in milliseconds that will be passed to ExifTool.',
+    }),
+  },
+  handler: async ({inFile, outFile, timeout}) => {
+    const exifTool = new ExifTool({ taskTimeoutMillis: timeout });
+    await exifTool.rewriteAllTags(inFile, outFile);
+    exifTool.end();
+  },
+});
+
+const fullMigrate = command({
+  name: 'google-photos-migrate-full',
+  args: {
+    takeoutDir: positional({
+      type: string,
+      displayName: 'takeout_dir',
+      description: 'The path to your "Takeout" directory.',
+    }),
+    timeout: option({
+      type: number,
+      defaultValue: () => 30000,
+      short: 't',
+      long: 'timeout',
+      description:
+        'Sets the task timeout in milliseconds that will be passed to ExifTool.',
+    }),
+  },
+  handler: async ({ takeoutDir, timeout }) => {
+    const errs: string[] = [];
+    if (!existsSync(takeoutDir)) {
+      errs.push('The specified takeout directory does not exist.');
+    }
+    if (errs.length !== 0) {
+      errs.forEach((e) => console.error(e));
+      process.exit(1);
+    }
+    if (await isEmptyDir(takeoutDir)) {
+      errs.push('The google directory is empty. Nothing to do.');
+    }
+    if (!(await isEmptyDir(`${takeoutDir}/Photos`))) {
```

Comment:
You should either create this directory, or document that it needs to be created manually.  It fails here if it doesn't already exist when you run it on a cleanly extracted takeout folder.

Reply:
It's also a little unusual you're expecting these to be created before you're called, bu then don't expect them to be passed in as arguments (like the `folderMigrate` `output` and `error` directories are).  I suspect you should be creating these yourself if they don't exist, rather than asking the user to create some "magic" hardcoded directories for you.